tools/block-generator/runner/runner.go

```diff
@@ -35,17 +35,21 @@ func init() {
 		Use:   "runner",
 		Short: "Run test suite and collect results.",
 		Long:  "Run an automated test suite using the block-generator daemon and a provided conduit binary. Results are captured to a specified output directory.",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error{
```

Comment:
Why `RunE`? I'm curious because I've never been able to figure out when it should be used. Last I checked it was poorly documented and the behavior was surprising for some reason or another.

Reply:
I'm replacing the usage of `Run` with `RunE` because of the following:

* There is no longer a general requirement to provide a connection string so I've removed `RunnerCmd.MarkFlagRequired("postgres-connection-string")`
* I want to assert that when the template is `postgres-exporter`, a connection string must be provided
* When the assertion fails, I return an error right away before continuing with the program logic, so that is where `RunE` comes in handy - letting me return a validation error